src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -3374,8 +3374,16 @@ void CodeGen::genFloatToIntCast(GenTree* treeNode)
     }
 
     genConsumeOperands(treeNode->AsOp());
+    regNumber tmpReg = rsGetRsvdReg();
```

Comment:
Does it need really need to be t6 or could you let the compiler assign any temp reg?

Reply:
I have tested GenTree::ExtractTempReg, and I got assert `genCountBits(availableSet) >= 1` with the `RBM_ALLINT` mask. Thus it seems like there is no temp reg available in such context. Do you have any clue why would it happened?
Reproducer: [sample.tar.gz](https://github.com/dotnet/runtime/files/13425842/sample.tar.gz)

Reply:
AFAIK you need to allocate it with buildInternalIntRegisterDefForNode and buildInternalRegisterUses in lsrarisc64.cpp. See GT_CMPXCHG for analogy.

Reply:
Right. We should prefer having LSRA indicate a temp register is required and then to extract that here, over forcing a particular to be used.

Reply:
Fixed.